crates/noirc_abi/src/lib.rs

```diff
@@ -218,11 +262,24 @@ impl Abi {
         Ok(encoded_value)
     }
 
+    /// Decode a `WitnessMap` into the types specified in the ABI.
+    pub fn decode_from_witness(&self, witness_map: &WitnessMap) -> Result<InputMap, AbiError> {
```

Comment:
This currently fails to decode a witness when:
a. The circuit has a return value
b. The return value's witness is not included as one of the inputs
c. The witness has not been solved yet.

I think this is fine for now until we address #633 and we can add special handling for this in a non-hacky way.

Reply:
Do you mean that the following will fail?

```rust
fn main(x : Field) -> pub Field {
let z = x + x;
z // return value here
}
```

Reply:
I've changed this so it will throw the error `Could not read witness value at index Witness(n), required by parameter "return"` instead of panicking.

Reply:
Makes sense -- The error is nicer UX